devnet-sdk/testing/systest/systest.go

```diff
@@ -107,41 +107,41 @@ func (h *basicSystemTestHelper) handlePreconditionError(t BasicT, err error) {
 	}
 }
 
-func (h *basicSystemTestHelper) SystemTest(t BasicT, f SystemTestFunc, validators ...PreconditionValidator) {
+func (h *basicSystemTestHelper) AcquireSystem(t BasicT, validators ...PreconditionValidator) (T, system.System) {
 	wt := NewT(t)
 	wt.Helper()
 
 	ctx, cancel := context.WithCancel(wt.Context())
-	defer cancel()
+	t.Cleanup(cancel)
```

Comment:
This is where I would be a bit hesitant with this approach - when does this `cancel` run exactly? 

And what are the implications for when tests are run in parallel (I am new to go so please feel free to discard my comments) - do we get a new `t` instance for every test, with `t.Cleanup` being only called once per each? What happens if we acquire two systems, does one `Cleanup` override the other one?

The advantage of the original approach here was that the lifecycle was limited to this function, if you expose & return the system, there are some more details to consider.

Reply:
`t.Cleanup` is the canonical way to clean up test resources in go tests.  You do get a new `t` for each test instance and multiple calls to `Cleanup` will all get run at the end of the test so it works fine if you acquire multiple systems still.  We should be leveraging the testing tools that go provides like this rather than trying to build it all ourselves, especially given the callback approach adds a bunch of extra boilerplate and complexity to the test code.

Basically using `defer` we avoid the need to understand `t.Cleanup` in this particular case (but it's used in so many other places understanding it is really table stakes), but every single test has to write multiple functions and understand this callback flow because the code no longer executes linearly.